### PR TITLE
feat: add all raptor tags

### PR DESCRIPTION
**Pull request by H4ckerxx44**
## Why?

I think I don't have to explain that the current tag system is mess.

How about previous gets a `tag` command which - instead of sending the content of the tag - just sends the link to the markdown file which then would contain all the info.

This way we are no longer limited to 2000 characters for tags which get a little longer and also we know who changed what in which tag and all. We also would have proper backups of the tag due to VCS.

The only disadvantage currently is that previous does not have a command like this. 

PS: This is only markdown files, no code changed.

### CODE REVIEW

**Review by Soheab — APPROVED** [review]

File: cogs/tags.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from nextcord.ext import commands
+
+class Tags(commands.Cog):
+    @commands.command()
+    async def tag(self, ctx, *, name: str):
+        """Grabs the Markdown file for a tag."""
+        branch_name = "master" # TODO: Dynamically grab current branch name via git
+        markdown_link = f"https://raw.githubusercontent.com/nextcord/previous/{branch_name}/assets/{name}.md"
+
+        async with aiohttp.ClientSession() as sess:
```
Comment (Soheab) [inline cogs/tags.py:12]: This should use the global session defined in main.py
https://github.com/nextcord/previous/blob/master/main.py#L92-L93

Comment (EmreTech) [reply cogs/tags.py:12]: Oh I didn't realize that existed. Mb

**Comment by DenverCoder1** [comment]
What will this do about dynamic tags that take additional arguments, for example?

**Comment by EmreTech** [comment]
> What will this do about dynamic tags that take additional arguments, for example?

Maybe we could implement a drop-in system where it uses a tag like this: `<arg1>`
That's just my idea, IDK what Dennis would do for that

**Comment by H4ckerxx44** [comment]
> What will this do about dynamic tags that take additional arguments, for example?

I would honestly just refactor them to some solid structure, for example the `examples application_commands` could be a single markdown file of `examples.md` with seperate sections, the tags need a rework anyways.

**Comment by Soheab** [comment]
> What will this do about dynamic tags that take additional arguments, for example?

Easier said than done but I'm thinking of writing a func pulls the content and formats the argument and content together and send it.

**Comment by EmreTech** [comment]
> > What will this do about dynamic tags that take additional arguments, for example?
> 
> Easier said than done but I'm thinking of writing a func pulls the content and formats the argument and content together and send it.

That's similar to what I was thinking
But then the problem is what's the point of this PR because the tag's contents is sent in chat anyways instead of being in a link.

So maybe we could pull the content, format the arguments, then send it to some other website that renders Markdown.

**Comment by Soheab** [comment]
> > > What will this do about dynamic tags that take additional arguments, for example?
> > 
> > 
> > Easier said than done but I'm thinking of writing a func pulls the content and formats the argument and content together and send it.
> 
> That's similar to what I was thinking But then the problem is what's the point of this PR because the tag's contents is sent in chat anyways instead of being in a link.
> 
> So maybe we could pull the content, format the arguments, then send it to some other website that renders Markdown.

Ahh right my bad, I forgot about the original purpose of doing this.. uh yeah that's annoying..

**Comment by koala9712** [comment]
Absolutely not shit idea: tags have tags/topics
right now, searching for tags is a mess, I don't know half of what those abbrieviations mean, and it'll become harder to read just the names of all the tags - Hence, tags could have "topics", so one could "search" (command) for say, topic `db` or `blocking` or whatever.

Heck, way better if the static ones (which do not require dynamic arguments)- such as message content stuff (hot topic nowadays 💀) or like images in embeds, go to the faq. This faq can further have such topic search or fuzzy search.

**Pull request closed without merging**